### DISCUSSION TITLE
Docker on Windows?

### DISCUSSION THREAD

**Discussion by Lka1988** [pine64/OpenPineBuds#98] [Q&A] [answered]
I'm posting this in Discussions as I don't think it has to do with the repo at all.

I'm pretty new when it comes to development. I figured I'd pick up a few of these Pine64 devices to learn... I'm no stranger to Docker, I run a small Proxmox-based homelab which hosts multiple VMs that run various Docker containers. Same with Git; I know enough to get by and can generally google what I don't know.

This, however, is proving to be a bit of a pain. I've been poking at this off and on for the last few months. I'm running Windows 10 and have the container running in Docker desktop, but I cannot figure out how to get `build.sh` to run. I had to install Git for Windows to get the repo cloned onto my PC, and was able to run `start_dev.sh` from there to spin up the container.

My issue is that when I try to run `build.sh` inside the container, I get this error: `/usr/bin/env: 'sh\r': No such file or directory`.

Should I be running this entire operation on a Linux machine instead? Or did I do something wrong? The Readme isn't very clear on where to run things, nor does it point to any references...

**Comment by Ralim** [comment, marked as answer]
When mixing a Linux docker with windows host it can get messy.

 Whats happened here is you likely cloned down the repository in windows got, which has re-written all the like endings to CRLF when Linux wants just LF.

The best solution for windows I've found is to use WSL, and do the git clone inside the that and also do docker etc inside it. Most tools should be WSL aware now (vscode, visual studio etc all are) and so you should be able to set the idea to use WSL instead of the windows command line

**Reply by Lka1988** [reply]
Ahh, ok. Looks like I need to do a bit more reading on this. I'm more than happy to spin up a Linux instance on a spare PC, but I enjoy the challenge for reasons beyond my own understanding...

Thank you.

**Reply by Ralim** [reply]
Absolutely it's a fantastic way to learn 😁

I think if you do it all in the WSL environment it should just work 🤣


Should be roughly)
1) install Ubuntu WSL
2) apt install build-essential and hit
3) git clone --recursive the repo
4) make targets to get the list of commands

You may need to change docker to work in WSL -- haven't tried in a Loong time

**Reply by Lka1988** [reply]
I've got Debian WSL installed already, but I'm still running into the same issue 🤔 I think it's #4 that I need to do, so I'll give that a shot when I'm back at my PC.
